### ISSUE TITLE
Encoder train on crawled face or FFHQ

### ISSUE THREAD

**Issue by PXThanhLam** [genforce/idinvert#42]
Thanks for your great work. I have run your repo and notice that, the inverted result on your example and image generate by pretrain stylegan generator is good, but when running on random image from the internet, the result is really bad. Is your encoder train on FFHQ or some random image from internet. In addition, the example image in your repo is random or take from your training set on encoder network. Thanks you.

**Comment by zhujiapeng** [comment]
Both the generator and encoder were trained using FFHQ datasets,  which have already been aligned. Thus, if you want to invert the image from the Web, make sure that you have aligned the image first.

**Comment by PXThanhLam** [comment]
Thanks for your answer, I will align the image before feeding into network.
Ps: In the perceptual model, did you try to calculate l2 loss in multi layer, instead of layer 22 only ( I calculate l2 loss in multi layer and notice that the result is much better )

**Comment by zhujiapeng** [comment]
We don't try it on multi-layers. Better reconstruction? or better manipulation? or something else?

**Comment by PXThanhLam** [comment]
Better Reconstruction, I notice that the mse loss between two image is much smaller when doing perceptual loss in multi-layers, and the reconstruct image look more "realistic". Here is some reconstruct image with and without multi-layer perceptual. ( with 1000 iteration)
Original Image:
![dicaprio_ori](https://user-images.githubusercontent.com/49019526/105606562-bb2e0900-5dcc-11eb-8069-eb04c1072c33.png)
Reconstruct with multi-layer perceptual loss:
![dicaprio_multi](https://user-images.githubusercontent.com/49019526/105606576-d8fb6e00-5dcc-11eb-80f1-a8936748a67b.png)
Reconstruct with original perceptual loss:
![dicaprio_single](https://user-images.githubusercontent.com/49019526/105606619-14963800-5dcd-11eb-9c7e-a2f48ba68ebb.png)
Ps : I have try to algin and the result is the same (Like what you see in example above), I currently try to expand your work with styleganv2, will inform you when the work is finish.

**Comment by zhujiapeng** [comment]
Directly apply this repo to the StyleGANv2 will not work

**Comment by ShenYujun** [comment]
Thanks for pointing this out. We will test it following your suggestion.
